src/Database/Schema/MysqlSchema.php

```diff
@@ -92,7 +92,7 @@ protected function _convertColumn($column)
             $precision = (int)$precision;
         }
 
-        if (in_array($col, ['date', 'time', 'datetime', 'timestamp'])) {
+        if (in_array($col, ['date', 'time', 'datetime', 'timestamp', 'year'])) {
             return ['type' => $col, 'length' => null];
```

Comment:
Once you have a schema object from reflection, you can use `$schema->createSql($table->connection())` to get the generated SQL.